### PR TITLE
Lookless file dialogs

### PR DESCRIPTION
**Pull request by Splitwirez**
## What does the pull request do?
Makes the Managed File Dialogs lookless, so they can be retemplated.

With Default theme:
Screenshot_20200903_193214

With Fluent theme:
Screenshot_20210423_152042
(formerly Screenshot_20200903_193005 )

## What is the current behavior?
Managed File Dialogs are UserControls, and thus cannot be readily customized

## What is the updated/expected behavior with this PR?
Increased cosmetic flexibility

## How was the solution implemented (if it's not obvious)?
It's kind of obvious, no?

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes
I suppose merging this would mean the dialogs would no longer "just work" in an Avalonia app which does not merge either of Avalonia's built-in themes - such an app would now have to provide its own templates.

### CODE REVIEW

File: src/Avalonia.Headless/mono_crash.2374905179.0.json
Comment (maxkatz6) [inline src/Avalonia.Headless/mono_crash.2374905179.0.json:1]: Committed by mistake

Comment (Splitwirez) [reply src/Avalonia.Headless/mono_crash.2374905179.0.json]: Should be gone now

**Comment by danwalmsley** [comment]
@Splitwirez im going to hold off merging this until after 0.10.0 release... so quite soon after that... 0.10.1 release.

Just because this control is so critical to many applications... we cant afford to break them or introduce a tiny subtle bug...

however I like the PR.. its very good and we will merge it after the release.

**Comment by Splitwirez** [comment]
> @Splitwirez im going to hold off merging this until after 0.10.0 release... so quite soon after that... 0.10.1 release.
> 
> Just because this control is so critical to many applications... we cant afford to break them or introduce a tiny subtle bug...
> 
> however I like the PR.. its very good and we will merge it after the release.

That makes perfect sense to me. I'll try to get the XAML cleaned up in the meantime.

**Comment by Splitwirez** [comment]
@danwalmsley So uh...now that 0.10.0 is out...has the time come for this PR?

**Comment by maxkatz6** [comment]
@danwalmsley @grokys do we want to merge it into 0.10.1 or wait for starting work on next big release (0.11 or 11.0)?

**Comment by Splitwirez** [comment]
@maxkatz6 @danwalmsley It just hit me earlier today that there actually _is_ a potential breaking change to come of this PR, albeit a rather...obscure one. I've just updated the initial PR post to explain what that breaking change would be. Mentioning both of you, to make sure this PR isn't merged without the breaking change being taken into account.

I am so sorry I didn't figure this out sooner...

**Comment by maxkatz6** [comment]
@Splitwirez I believe this breaking change is pretty expected. Still good to mention it, thanks.
Meanwhile, can you fix conflicts?

**Comment by Splitwirez** [comment]
...update: I am currently banging my head against the wall trying to figure out some technical difficulties regarding compiling Avalonia ~~while also coming face to face with how git-illiterate I am~~. Hopefully this won't take *too* much longer...?

**Comment by Splitwirez** [comment]
@maxkatz6 Okay, the conflicts are resolved, icons updated, and the Fluent theme styles actually look...well, Fluent now. I say it's ready to be merged...unless you folks want further changes made, of course.

File: src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml:275]: I think this padding can be a bit smaller, like 30 instead of 34 or so. Also ListBox width from 250 to ~240.

File: src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml:177]: Initial window size feels too big for me, as well as Name column is way bigger than average file name. Probably Window size can be changed to be smaller, and column width updated to fit everything.

File: src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml
Comment (Splitwirez) [reply src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml]: Done and done :+1:

File: src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml
Comment (Splitwirez) [reply src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml]: Name column default width is now 275, which brings the entirety of the other columns to just within the default window bounds.

File: src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json
Comment (Splitwirez) [reply src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json:3543]: I didn't touch that...in fact, I don't even know what it is...

Comment (maxkatz6) [reply src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json:3543]: NPM packages lock file, it's safe to just revert changes in it.

Comment (Splitwirez) [reply src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json:3543]: ...how exactly do I do that here?

Comment (maxkatz6) [reply src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json:3543]: Easy way - just copy file without changes from the repo and replace in your local directory, commit.
Harder way - rebase, discard changes in this file on the specific commit, where it was changed.

Either way is fine.

Comment (Splitwirez) [reply src/Avalonia.DesignerSupport/Remote/HtmlTransport/webapp/package-lock.json]: Okay, done and done :+1:

**Review by maxkatz6 — APPROVED** [review]
Thanks!

**Pull request merged**
